api/cpp/docs/cmake_reference.md

```diff
@@ -0,0 +1,39 @@
+# CMake Reference
+
+## `slint_target_sources`
+
+```
+slint_target_sources(<target> <files>....)
+```
+
+Similar to the builtin cmake [target_sources](https://cmake.org/cmake/help/latest/command/target_sources.html) macro,
+this can be used to tell cmake about the .slint files to use.
+The macro will take care of running the slint-compiler to convert .slint file to a .h file in the build directory,
+while making sure that the include path are approprietly set so that you can include it.
+
```

Comment:
```suggestion
The function takes care of running the slint-compiler to convert `.slint` files to `.h` files in the build directory,
and extend  the include directories of your target so that the generated file is found when including it in your application.

```